src/migration/1688029905946-PostFlagsSync.ts

```diff
@@ -0,0 +1,30 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class PostFlagsSync1688029905946 implements MigrationInterface {
+    name = 'PostFlagsSync1688029905946'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{sentAnalyticsReport}\', \'true\'::jsonb) WHERE "sentAnalyticsReport" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{sentAnalyticsReport}\', \'false\'::jsonb) WHERE "sentAnalyticsReport" = FALSE')
+
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{banned}\', \'true\'::jsonb) WHERE "banned" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{banned}\', \'false\'::jsonb) WHERE "banned" = FALSE')
+
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{deleted}\', \'true\'::jsonb) WHERE "deleted" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{deleted}\', \'false\'::jsonb) WHERE "deleted" = FALSE')
+
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{private}\', \'true\'::jsonb) WHERE "private" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{private}\', \'false\'::jsonb) WHERE "private" = FALSE')
+
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{visible}\', \'true\'::jsonb) WHERE "visible" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{visible}\', \'false\'::jsonb) WHERE "visible" = FALSE')
+
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{showOnFeed}\', \'true\'::jsonb) WHERE "showOnFeed" = TRUE')
+        await queryRunner.query('UPDATE post SET flags = jsonb_set(flags, \'{showOnFeed}\', \'false\'::jsonb) WHERE "showOnFeed" = FALSE')
```

Comment:
We should turn it into one query that updates all the relevant posts where flags are not set to the default value. We also have to make sure it will not trigger a CDC update. I think there's an SQL statement that we can run before the update query to disable it for this transaction. But I don't remember precisely the command.

Reply:
> We should turn it into one query that updates all the relevant posts where flags are not set to the default value.

@idoshamun not sure about what you mean here, so you want all the updates in single `UPDATE` statement?

Reply:
Exactly! But we also have to take into account the CDC side effect to it. If you update the post table it will generate thousands of CDC messages with no actual value. We can run an SQL statement to disable it for the transaction. Anyway happy to go on a call if needed.